WordPress/gutenberg
# Components: Refactor `Tooltip` tests to `@testing-library/react`

Part of #42753

## What?
Refactoring of the Tooltip component is being considered in #42753.
As a first step, I rewritten tests to `@testing-library/react`.

## How?
I straightforwardly replacd `enzyme` tests with `@testing-library/react`, with a few exceptions, based on the following post:

- [Migrate from Enzyme | Testing Library](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme/)
- [Common mistakes with React Testing Library](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen)

## Testing Instructions
~Verify tests pass: `npm run test-unit ./packages/components/src/tooltip/test/index.js`~
Verify tests pass: `npm run test:unit ./packages/components/src/tooltip/test/index.js`

## Screenshots or screencast
